backend/src/channels/channels.service.ts

```diff
@@ -40,9 +52,19 @@ export class ChannelsService {
       } else {
         channel.password = undefined;
       }
+      await this.usersRepository.findOneByOrFail({
+        id: +createChannelDto.userId,
+      });
       channel.name = createChannelDto.name;
+      ret = await this.channelsRepository.save(channel);
+      role = MembershipRoleType.OWNER;
     }
-    return await this.channelsRepository.save(channel);
+    this.membershipsService.create({
+      userId: userId,
+      role: role,
+      channelId: channel.id,
+    });
```

Comment:
When the `channel.type` is `DIRECT`, it seems that there is no `save`-ing in the database and that the code flow makes 2 identical calls to `this.membershipsService.create`, one in the main function, one `checkDirectUsers`. Is that on purpose? If not, I would remove any call to `save` or `create` in `checkDirectUsers` or change the function name.

Reply:
That's right, also it only created one of the two memberships needed. Changing it.